controllers/glanceapi_controller.go

```diff
@@ -331,6 +336,45 @@ func (r *GlanceAPIReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		}
 		return nil
 	}
+	tpFn := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			oldObj := e.ObjectOld.(*topologyv1.Topology)
+			newObj := e.ObjectNew.(*topologyv1.Topology)
+			// Compare spec
+			return !equality.Semantic.DeepEqual(oldObj.Spec, newObj.Spec)
+		},
+	}
+
+	topologyFn := func(_ context.Context, o client.Object) []reconcile.Request {
+		result := []reconcile.Request{}
+
+		// get all GlanceAPIs CRs
+		glanceAPIs := &glancev1.GlanceAPIList{}
+		listOpts := []client.ListOption{
+			client.InNamespace(o.GetNamespace()),
+		}
+		if err := r.Client.List(context.Background(), glanceAPIs, listOpts...); err != nil {
+			r.Log.Error(err, "Unable to retrieve GlanceAPI CRs %w")
+			return nil
+		}
+
+		for _, cr := range glanceAPIs.Items {
+			if cr.Spec.Topology != nil {
+				if o.GetName() == cr.Spec.Topology.Name {
+					name := client.ObjectKey{
+						Namespace: o.GetNamespace(),
+						Name:      cr.Name,
+					}
+					r.Log.Info(fmt.Sprintf("Topology %s is used by GlanceAPI CR %s", o.GetName(), cr.Name))
+					result = append(result, reconcile.Request{NamespacedName: name})
+				}
+			}
+		}
+		if len(result) > 0 {
+			return result
+		}
+		return nil
+	}
```

Comment:
wondering if we should align all those watch funcs and instead use IndexField where we have named objects to watch and use the findObjectsForSrc. I could see indexing the field is more efficient, but can not confirm.

Reply:
Let me try this one and I get back with an update